tests/test_constrained.py

```diff
@@ -1896,6 +1896,15 @@ def test_add_linear_from(self):
 
         self.assertEqual(cqm.objective.linear, {'x': 3})
 
+    def test_add_quadratic_from(self):
```

Comment:
It would be good to also have tests for:
* input as mapping
* missing variables (ones that have not already been added to the CQM).